### PR TITLE
[elixir+python] feat: refactor for generic ingestion and ODS ingestion

### PR DESCRIPTION
**Pull request by grejdi-mbta**
For reference: [Asana](https://app.asana.com/0/1201701089427502/1202211702864023)

This PR re-adds support for ODS ingestion as indicated in this [ADR](https://github.com/mbta/data_platform/blob/main/doc/adr/0005-refactoring-ingestion-for-all-data.md).

### CODE REVIEW

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex [hunk 1]
```diff
@@ -97,7 +98,9 @@ defmodule ExCubicIngestion.Schema.CubicLoad do
         Enum.reduce(filters, __MODULE__, fn {s3_key, s3_modified}, query ->
           # query
           from(load in query,
-            or_where: load.s3_key == ^s3_key and load.s3_modified == ^s3_modified
+            or_where:
+              is_nil(load.deleted_at) and load.s3_key == ^s3_key and
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex:102]: Added deleted check.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_ods_load_snapshot.ex [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    field(:snapshot, :utc_datetime)
+
+    field(:deleted_at, :utc_datetime)
+
+    timestamps(type: :utc_datetime)
+  end
+
+  @spec get_by!(Keyword.t() | map(), Keyword.t()) :: t() | nil
+  def get_by!(clauses, opts \\ []) do
+    # @todo add is_nil(deleted_at) somehow
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_ods_load_snapshot.ex:42]: @paulswartz was trying to add deleted check here with a `Keyword.put(clauses, deleted_at, nil)`, but Ecto doesn't like comparisons to `nil`, and was giving me an error. Let me know if you have any tips.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_table.ex [hunk 3]
```diff
@@ -29,7 +30,6 @@ defmodule ExCubicIngestion.Schema.CubicTable do
 
   schema "cubic_tables" do
     field(:name, :string)
-    # @todo make this unique
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_table.ex:32]: Done with an index in the migration.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/start_ingestion.ex [hunk 4]
```diff
@@ -51,16 +52,32 @@ defmodule ExCubicIngestion.StartIngestion do
...
+      |> Enum.filter(&String.starts_with?(&1.s3_key, "cubic/ods_qlik/"))
+      |> Enum.each(&CubicOdsLoadSnapshot.update_ods_snapshot(&1))
+
+    # start ingestion
+    ready_loads
     |> chunk_loads()
     |> Enum.each(&process_loads/1)
   end
 
+  @spec chunk_loads([CubicLoad.t()]) :: [[CubicLoad.t(), ...]]
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_cubic_ingestion/start_ingestion.ex:75]: I just moved this up, and will replace with [the other PR](https://github.com/mbta/data_platform/pull/42) after approval.

File: ex_cubic_ingestion/priv/repo/migrations/202205110144359_add_sample_and_use_transaction_tables.exs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  alias ExCubicIngestion.Repo
+  alias ExCubicIngestion.Schema.CubicTable
+  alias ExCubicIngestion.Schema.CubicOdsTableSnapshot
+
+  def up do
+    edw_sample_table_rec = Repo.insert!(%CubicTable{
+      name: "cubic_ods_qlik__edw_sample",
+      s3_prefix: "cubic/ods_qlik/EDW.SAMPLE/"
+    })
+    Repo.insert!(%CubicOdsTableSnapshot{
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/priv/repo/migrations/202205110144359_add_sample_and_use_transaction_tables.exs:13]: Will update documentation in Notion after approval.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_ods_load_snapshot.ex (hunk 2 above)
Comment (paulswartz) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_ods_load_snapshot.ex:42]: I think you want something like this:
```
defp not_deleted do
  from load in __MODULE__, where: is_nil(deleted_at)
end

def get_by!(clauses, opts \\ []) do
  Repo.get_by!(not_deleted(), clauses, opts)
end
```

[Ecto's docs](https://hexdocs.pm/ecto/Ecto.Query.html#module-composition) have more on composing queries as well.

**Comment by grejdi-mbta** [comment]
@paulswartz I know these PRs are rather large, but I wanted to take some time to address testing to the best of my ability. All of it has also been tested with real infrastructure utilizing the 'local' bucket, as well my own Glue database and tables.

**Comment by paulswartz** [comment]
> @paulswartz I know these PRs are rather large, but I wanted to take some time to address testing to the best of my ability. All of it has also been tested with real infrastructure utilizing the 'local' bucket, as well my own Glue database and tables.

Should I be seeing data in s3://mbta-ctd-dataplatform-local/ggjura/springboard/ after your testing? I don't see the `USE_TRANSACTION` data in `incoming` so it appears to have been processed, but I don't see any outputs.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/start_ingestion.ex (hunk 4 above)
Comment (paulswartz) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/start_ingestion.ex:75]: The other PR was already approved.

**Comment by grejdi-mbta** [comment]
> Should I be seeing data in s3://mbta-ctd-dataplatform-local/ggjura/springboard/ after your testing? I don't see the `USE_TRANSACTION` data in `incoming` so it appears to have been processed, but I don't see any outputs.

I just wiped it out to do a bigger test with all the tables.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/start_ingestion.ex (hunk 4 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/start_ingestion.ex:75]: I meant after the approval of this PR. Thanks.

**Comment by grejdi-mbta** [comment]
> question: should `py_cubic_ingestion/py_cubic_ingestion.zip` be ignored rather than checked in?

It's an artifact of deployment, so it should be ignored. Are you thinking about something else?

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex (hunk 1 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex:102]: Ah.. will try that.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/process_incoming_test.exs [hunk 6]
```diff
@@ -29,24 +30,50 @@ defmodule ExCubicIngestion.ProcessIncomingTest do
...
+
+      dmap_table_id = dmap_table.id
+
+      ods_table =
+        Repo.insert!(%CubicTable{
+          name: "cubic_ods_qlik__sample",
+          s3_prefix: "cubic/ods_qlik/SAMPLE/"
+        })
+
+      ods_table_id = ods_table.id
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/process_incoming_test.exs:48]: Only one test that uses it for now, so didn't feel the need to do it.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs [hunk 7]
```diff
@@ -7,59 +7,61 @@ defmodule ExCubicIngestion.Schema.CubicLoadTest do
...
+      assert {:ok, []} == CubicLoad.insert_new_from_objects_with_table([], dmap_table)
     end
   end
 
   describe "get_by_objects/1" do
     test "getting records just added by providing the list we added from", %{
-      table: table,
-      load_objects: load_objects
+      dmap_table: dmap_table,
+      dmap_load_objects: dmap_load_objects
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs:51]: You're right. I was thinking about it as I was doing it. Just wanted to have the variable names represent accurate information, but in this case and for indicating that the test is generic, the variable names should be generic.

**Comment by paulswartz** [comment]
> > question: should `py_cubic_ingestion/py_cubic_ingestion.zip` be ignored rather than checked in?
> 
> It's an artifact of deployment, so it should be ignored. Are you thinking about something else?

It looks like it's checked in to this branch:
Screen Shot 2022-05-16 at 10 18 17

File: ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs [hunk 8]
```diff
@@ -152,50 +154,40 @@ defmodule ExCubicIngestion.Schema.CubicLoadTest do
...
-               Enum.map(
-                 CubicLoad.get_many_with_table([first_load_rec.id]),
-                 fn {load_rec, table_rec} ->
-                   {load_rec.id, table_rec.id}
-                 end
+      # sort actual recs, as sometimes they might come back out of order
+      assert Enum.map(new_load_recs, &{&1, dmap_table}) ==
+               Enum.sort(
+                 CubicLoad.get_many_with_table(new_load_rec_ids),
+                 &(elem(&1, 0).id < elem(&2, 0).id)
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs:171]: I'll change it. I also was on the fence about this.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/start_ingestion_test.exs [hunk 9]
```diff
@@ -3,32 +3,62 @@ defmodule ExCubicIngestion.StartIngestionTest do
...
+    dmap_load_1 =
+      Repo.insert!(%CubicLoad{
+        table_id: dmap_table.id,
+        status: "ready",
+        s3_key: "cubic/dmap/sample/20220101.csv",
+        s3_modified: ~U[2022-01-01 20:49:50Z],
+        s3_size: 197
+      })
+
+    ods_load_1 =
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/start_ingestion_test.exs:47]: I guess not. I will update.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex (hunk 1 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex:112]: Substituting `not_deleted()` for `__MODULE__` didn't work. Will tweak some more, but wanted to move forward.

File: py_cubic_ingestion/tests/test_job_helpers.py [hunk 10]
```diff
@@ -269,15 +282,49 @@ def test_write_parquet(spark_session: SparkSessionType, tmp_path: str) -> None:
...
+        ("test_5", "2022-01-05", "snapshot_1", "identifier_1"),
+        ("test_6", "2022-01-06", "snapshot_1", "identifier_1"),
     ]
-    expected_df = spark_session.createDataFrame(expected_data, ["name", "date", "identifier"])
+    expected_df = spark_session.createDataFrame(expected_data, ["name", "date", "snapshot", "identifier"])
+
+    assert_equal_collections(parquet_df.collect(), expected_df.collect())
+
+
+def test_write_parquet_without_partition_columns(spark_session: SparkSessionType, tmp_path: str) -> None:
```
Comment (grejdi-mbta) [inline py_cubic_ingestion/tests/test_job_helpers.py:302]: Extra test in case Spark changes its API to not accept an empty partition list.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex [hunk 11]
```diff
@@ -123,10 +126,9 @@ defmodule ExCubicIngestion.Schema.CubicLoad do
 
   @spec get_status_ready :: [t()]
   def get_status_ready do
-    # @todo add deleted filter
     query =
       from(load in __MODULE__,
-        where: load.status == "ready",
+        where: is_nil(load.deleted_at) and load.status == "ready",
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_load.ex]: Done.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_table.ex [hunk 12]
```diff
@@ -51,15 +56,15 @@ defmodule ExCubicIngestion.Schema.CubicTable do
...
+      # strip any change tracking suffix (in ODS)
       without_change_tracking =
         prefixes
         |> MapSet.new(&String.replace_suffix(&1, "__ct/", "/"))
         |> Enum.to_list()
 
       query =
         from(table in __MODULE__,
-          where: table.s3_prefix in ^without_change_tracking
+          where: is_nil(table.deleted_at) and table.s3_prefix in ^without_change_tracking
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_table.ex]: Done.

[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/schema/cubic_table.ex]: Updated in other places as well.

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/workers/ingest.ex [hunk 13]
```diff
@@ -108,17 +109,33 @@ defmodule ExCubicIngestion.Workers.Ingest do
...
        GLUE_DATABASE_INCOMING: glue_database_incoming,
        S3_BUCKET_INCOMING: bucket_incoming,
        S3_BUCKET_PREFIX_INCOMING: prefix_incoming,
        S3_BUCKET_SPRINGBOARD: bucket_springboard,
        S3_BUCKET_PREFIX_SPRINGBOARD: prefix_springboard
-     }), Jason.encode!(%{loads: loads})}
+     }),
+     Jason.encode!(%{
+       generic_loads: generic_loads,
+       ods_loads: ods_loads_with_snapshot
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/ex_cubic_ingestion/workers/ingest.ex]: I have now combined them as suggested.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs (hunk 7 above)
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs]: Fixed.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs (hunk 8 above)
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs]: Updated.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/start_ingestion_test.exs (hunk 9 above)
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/start_ingestion_test.exs]: Updated.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/ingest_test.exs [hunk 14]
```diff
@@ -3,34 +3,233 @@ defmodule ExCubicIngestion.Workers.IngestTest do
...
+            "snapshot" => Calendar.strftime(ods_snapshot, "%Y%m%dT%H%M%SZ")
+          }
+        ]
+      }
+
+      assert expected_input_with_ods["generic_loads"] ==
+               actual_input_with_ods_decoded["generic_loads"]
+
+      assert expected_input_with_ods["ods_loads"] ==
+               Enum.sort(actual_input_with_ods_decoded["ods_loads"], &(&1["id"] < &2["id"]))
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/workers/ingest_test.exs]: No longer needed to separate, as now combined into one list, and then test feels accurate as it is now.

File: py_cubic_ingestion/lib/py_cubic_ingestion/ingest_incoming.py [hunk 15]
```diff
@@ -35,20 +35,40 @@ def run() -> None:
...
+        # add partition columns
+        table_spark_df = job_helpers.df_with_partition_columns(
+            table_df.toDF(),  # convert to spark df so we can use the withColumn functionality
+            [("snapshot", load["snapshot"]), ("identifier", os.path.basename(load["s3_key"]))],
+        )
+
+        # write out to springboard bucket using the same prefix as incoming
+        job_helpers.write_parquet(
+            table_spark_df,
+            ["snapshot", "identifier"],
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply py_cubic_ingestion/lib/py_cubic_ingestion/ingest_incoming.py]: Done. I also moved `df_with_partition_columns()` within `job_helpers.write_parquet()`, making the `ingest_incoming.py` even smaller.

File: ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs [hunk 16]
```diff
@@ -0,0 +1,463 @@
+defmodule ExCubicIngestion.Repo.Migrations.AddRestOdsTableRecords do
```
Comment (paulswartz) [inline ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs:1]: question: should this be in this PR? I thought there was a separate PR for adding the additional tables.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs [hunk 17]
```diff
@@ -3,66 +3,103 @@ defmodule ExCubicIngestion.Workers.ArchiveTest do
...
+    test "getting destination key for ODS load", %{
+      ods_snapshot: ods_snapshot,
+      ods_load: ods_load
+    } do
+      expected_destination_key =
+        Enum.join([
+          Path.dirname(ods_load.s3_key),
+          '/snapshot=#{Calendar.strftime(ods_snapshot, "%Y%m%dT%H%M%SZ")}/',
+          Path.basename(ods_load.s3_key)
+        ])
```
Comment (paulswartz) [inline ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs:100]: suggestion: this test appears to be pretty much copying the implementation of the function. Given that all of this is static data, could you do something like:
```elixir
expected_destination_key = "cubic/ods/snapshot=YYYYMMDDTHHMMSSZ/LOAD000001.csv.gz"
```

File: py_cubic_ingestion/lib/py_cubic_ingestion/job_helpers.py [hunk 18]
```diff
@@ -32,13 +32,13 @@ def parse_args(env_arg: str, input_arg: str) -> typing.Tuple[dict, dict]:
     --------
     >>> parse_job_arguments(
     ...   '{"GLUE_DATABASE_INCOMING": "db","S3_BUCKET_INCOMING": "incoming","S3_BUCKET_SPRINGBOARD": "springboard"}',
-    ...   '{"loads":[{"s3_key":"s3/prefix/key","snapshot":"20220101","table_name":"table"}]}'
+    ...   '{"loads": [...], "partition_columns": [...]}'
```
Comment (paulswartz) [inline py_cubic_ingestion/lib/py_cubic_ingestion/job_helpers.py:35]: question: I thought `partition_columns` were a key within the `load` objects, not a separate top-level argument?

Comment (grejdi-mbta) [reply py_cubic_ingestion/lib/py_cubic_ingestion/job_helpers.py:35]: Yes. I'll update.

File: ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs (hunk 16 above)
Comment (paulswartz) [reply ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs:1]: What's the overlap with #43 which is still in draft? And isn't there a lot of Terraform/schema work needed before we add additional tables here?

Comment (grejdi-mbta) [reply ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs:1]: I have the [TF PR in draft](https://github.com/mbta/devops/pull/375) in the DevOps repo. Need to make slight modifications to the paths, but it is also ready. I don't plan on deploying this until that is out.
Regarding the overlap. This migration differs from #43 in regards to the different prefix, and the additional `cubic_ods_table_snapshot` record that needs to be inserted for each table.

Comment (paulswartz) [reply ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs:1]: This is a big refactor: I think it would be better to have it out and running before we add lots more tables.

As is, won't we run into the issue discussed in #46 as well?

File: ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs [hunk 19]
```diff
@@ -7,39 +7,39 @@ defmodule ExCubicIngestion.Schema.CubicLoadTest do
...
-               Enum.map(new_load_recs, fn new_load_rec ->
-                 %{
-                   status: new_load_rec.status,
-                   s3_key: new_load_rec.s3_key,
-                   s3_modified: new_load_rec.s3_modified,
-                   s3_size: new_load_rec.s3_size
-                 }
-               end)
+      assert Enum.map(load_objects, &%{s3_key: &1[:key]}) ==
+               Enum.map(new_load_recs, &%{s3_key: &1.s3_key})
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/schema/cubic_load_test.exs]: Updated.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs (hunk 17 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs]: Updated.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/ingest_test.exs [hunk 20]
```diff
@@ -3,34 +3,181 @@ defmodule ExCubicIngestion.Workers.IngestTest do
...
+                "value" => Calendar.strftime(ods_snapshot, "%Y%m%dT%H%M%SZ")
+              },
+              %{"name" => "identifier", "value" => Path.basename(ods_load_2.s3_key)}
+            ]
+          }
+        ]
+      }
+
+      assert expected_input["loads"] ==
+               Enum.sort(actual_input_decoded["loads"], &(&1["id"] < &2["id"]))
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/workers/ingest_test.exs]: Updated.

File: py_cubic_ingestion/lib/py_cubic_ingestion/job_helpers.py (hunk 18 above)
Comment (grejdi-mbta) [reply py_cubic_ingestion/lib/py_cubic_ingestion/job_helpers.py]: Updated.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs [hunk 21]
```diff
@@ -3,66 +3,41 @@ defmodule ExCubicIngestion.Workers.ArchiveTest do
...
-        Repo.insert!(%CubicLoad{
-          status: "ready",
-          table_id: new_table_rec.id,
-          s3_key: "#{new_table_rec.s3_prefix}LOAD1.csv",
-          s3_modified: created,
-          s3_size: 0
-        })
+  setup do
+    TestFixtures.setup_tables_loads()
+  end
```
Comment (paulswartz) [inline ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs:13]: suggestion(non-blocking):
```suggestion
import ExCubicIngestion.TestFixtures, only: [:setup_tables_loads]

setup :setup_tables_loads
```

**Review by paulswartz — APPROVED** [review]
🍰

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs (hunk 21 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs:13]: Neat!

Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/ex_cubic_ingestion/workers/archive_test.exs]: Needed to modify a bit to get right. See commit: https://github.com/mbta/data_platform/pull/45/commits/0e1feebbbf7580f47e546c1e1334fe675a435974

File: ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs (hunk 16 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/priv/repo/migrations/20220513171055_add_ods_table_records.exs]: I have removed the migration for adding all the tables and now just adds `sample` and `use_transaction`. See commit: https://github.com/mbta/data_platform/pull/45/commits/0e1feebbbf7580f47e546c1e1334fe675a435974

As is, it will run into the issue being addressed in #46.

File: ex_cubic_ingestion/priv/repo/migrations/202205110144359_add_sample_and_use_transaction_tables.exs (hunk 5 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/priv/repo/migrations/202205110144359_add_sample_and_use_transaction_tables.exs]: Updated.
